LB_DSC_19thApril_WKDY_MR_2024

Mohit Chaudhary 7011883015

Q1. Declare a integer value and store it in a variable.
Check the type and print the id of the same.

In [23]:
a = 20       #Declaring a integer value and storing it in a variable. 

print('The type of variable is:',type(a))   #Checking the type of variable.
print('The id of variable is:',id(a))       #Checking the id of variable

The type of variable is: <class 'int'>
The id of variable is: 140710734042504


Q2.  Take one integer value between 0 - 256.
Assign it to two different variables.
Check the id of both the variables. It should come the same. Check why?

In [8]:
b = 25            #Taking one integer value
c = 25            #Assigning it to another variable.


print('The id of first variable is:',id(b))   #Checking the id of integer a
print('The id of Second variable is:',id(c))  #Checking the id of integer b

The id of first variable is: 140710508926504
The id of Second variable is: 140710508926504


As we can see that both first and second variable is having same id because both variables has same value which is stored in memory location '140710508926504', this is called memory reusibility.

Q3. Take one int value either less than -5 or greater than 256.
Assign it to two different variables.
Check the id of both the variables. It should come different.Check why?

In [16]:
b = -15            #Taking one integer value
c = -15            #Assigning it to another variable.


print('The id of first variable is:',id(b))   #Checking the id of integer a
print('The id of Second variable is:',id(c))  #Checking the id of integer b

The id of first variable is: 1872390009424
The id of Second variable is: 1872390009264


As we can see that both first and second variable is having different id because both variables has same value but not in the range (-5 to 256) so the memory reusibility is not applicable beyond this range.

Q4.  Arithmetic Operations on integers
Take two different integer values.
Store them in two different variables.
Do below operations on them:-
Find sum of both numbers
Find difference between them
Find the product of both numbers.
Find value after dividing first num with second number
Find the remainder after dividing first number with second number Find the quotient after dividing first number with second number Find the result of the first num to the power of the second number.

In [24]:
a1 = int(input('Enter any integer number: '))      #Taking first value from user.
a2 = int(input('Enter another integer number: '))  #Taking second value from user.

print('sum of first and second value is', a1 + a2)                         #Getting the sum of both values
print('Difference of first and second value is', a1 - a2)                  #Finding the differce between both values
print('Product of first and second value is', a1 * a2)                     #Finding the product of both values.
print('Division of first with second value is', a1 / a2)                   #dividing first value with second value
print('Quotient after dividing first value with second value', a1 // a2)   #Quotient after dividing first value with second value
print('Remainder after dividing first value with second value', a1 % a2)   #remainder after dividing first value with second value
print('first value to the power of second value is', a1 ** a2)             #first value to the power of second value.


Enter any integer number:  10
Enter another integer number:  5


sum of first and second value is 15
Difference of first and second value is 5
Product of first and second value is 50
Division of first with second value is 2.0
Quotient after dividing first value with second value 2
Remainder after dividing first value with second value 0
first value to the power of second value is 100000


Q5. Comparison Operators on integers
Take two different integer values.
Store them in two different variables.
Do below operations on them:-
   Compare se two numbers with below operator:-        Greater than, '>'
       Smaller than, '<'
       Greater than or equal to, '>='
       Less than or equal to, '<='
Observe their output(return type should be boolean)Q. Comparison Operators on integers
Take two different integer values.
Store them in two different variables.
Do below operations on them:-
   Compare se two numbers with below operator:-        Greater than, '>'
       Smaller than, '<'
       Greater than or equal to, '>='
       Less than or equal to, '<='
Observe their output(return type should be boolean)

In [27]:
b1 = int(input('Enter any integer number: '))      #Taking first value from user.
b2 = int(input('Enter another integer number: '))  #Taking second value from user.

print('Is first value Greater than second value:', b1 > b2)               #Greater than, '>'
print('Is first value Less than second value:', b1 < b2)                  #less than, '<'
print('Is first value Less than or equal to second value:', b1 <= b2)     #Greater than or equal to, '>='
print('Is first value Greater than or equal to second value', b1 >= b2)   #Less than or equal to, '<='


Enter any integer number:  25
Enter another integer number:  10


Is first value Greater than second value: True
Is first value Less than second value: False
Is first value Less than or equal to second value: False
Is first value Greater than or equal to second value True


As we can observe that the output we have obtained in all scenarios is Boolean (True/False), this is because comparison operator always gives us Either True or False

Q6. Equality Operator
Take two different integer values.
Store them in two different variables. Equate them using equality operators (==, !=) Observe the output(return type should be boolean)Q. Equality Operator
Take two different integer values.
Store them in two different variables. Equate them using equality operators (==, !=) Observe the output(return type should be boolean)

In [31]:
c1 = int(input('Enter any integer number: '))            #Taking first value from user.
c2 = int(input('Enter another integer number: '))        #Taking second value from user.
print('Checking if values are equal using == :,', c1 == c2)        #Checking if values are equal
print('Checking if values are different using != :', c1 != c2)     #Checking if values are different


Enter any integer number:  20
Enter another integer number:  25


Checking if values are equal using == :, False
Checking if values are different using != : True


As we can observe that the output we have obtained in all scenarios is Boolean (True/False), this is because Equality operator always gives us Either True or False

Q7. Logical operators
Observe the output of below codeCross check the output manually
print(10 and 20)
#----------------------------------------->Output is 20
print(0 and 20)
#----------------------------------------->Output is 0
print(20 and 0)
#----------------------------------------->Output is 0
print(0 and 0)
#----------------------------------------->Output is 0
print(10 or 20)
#----------------------------------------->Output is 10
print(0 or 20)
#----------------------------------------->Output is 20
print(20 or 0)
#----------------------------------------->Output is 20
print(0 or 0)
#----------------------------------------->Output is 0
print(not 10)
#----------------------------------------->Output is False
print(not 0)
#----------------------------------------->Output is TrueQ. Logical operators
Observe the output of below codeCross check the output manually
print(10 and 20)
#----------------------------------------->Output is 20
print(0 and 20)
#----------------------------------------->Output is 0
print(20 and 0)
#----------------------------------------->Output is 0
print(0 and 0)
#----------------------------------------->Output is 0
print(10 or 20)
#----------------------------------------->Output is 10
print(0 or 20)
#----------------------------------------->Output is 20
print(20 or 0)
#----------------------------------------->Output is 20
print(0 or 0)
#----------------------------------------->Output is 0
print(not 10)
#----------------------------------------->Output is False
print(not 0)
#----------------------------------------->Output is True

In [38]:
#Lets check the output manually:

print(10 and 20)       
print(0 and 20)        
print(20 and 0)        
print(0 and 0)         

print(10 or 20)       
print(0 or 20)         
print(20 or 0)         
print(0 or 0)          

print(not 10)         
print(not 0)           

20
0
0
0
10
20
20
0
False
True


Q8. Bitwise Operators
Do below operations on the values provided below:-
   Bitwise and(&) -----------------------------------------> 10, 20
-------> Output is 0
   Bitwise or(|)  -----------------------------------------> 10, 20
-------> Output is 30
   Bitwise(^)     -----------------------------------------> 10, 20 -------> Output is 30
   Bitwise negation(~) ------------------------------------> 10 -------> Output is -11
   Bitwise left shift  ------------------------------------> 10,2 -------> Output is 40
   Bitwise right shift ------------------------------------> 10,2 -------> Output is 2
Cross check the output manuallyQ. Bitwise Operators
Do below operations on the values provided below:-
   Bitwise and(&) -----------------------------------------> 10, 20
-------> Output is 0
   Bitwise or(|)  -----------------------------------------> 10, 20
-------> Output is 30
   Bitwise(^)     -----------------------------------------> 10, 20 -------> Output is 30
   Bitwise negation(~) ------------------------------------> 10 -------> Output is -11
   Bitwise left shift  ------------------------------------> 10,2 -------> Output is 40
   Bitwise right shift ------------------------------------> 10,2 -------> Output is 2
Cross check the output manually

In [42]:
#Lets check the output manually:

print(10&20)
print(10|20)
print(10^20)
print(~10)
print(10<<2)
print(10>>2)

0
30
30
-11
40
2


Q9. What is the output of expression inside print statement. Cross check before running the program.a = 10
b = 10
print(a is b)          #True or False?
print(a is not b)      #True or False?
a = 1000
b = 1000
print(a is b)          #True or False?
print(a is not b)      #True or False?Q. What is the output of expression inside print statement. Cross check before running the program.a = 10
b = 10
print(a is b)          #True or False?
print(a is not b)      #True or False?
a = 1000
b = 1000
print(a is b)          #True or False?
print(a is not b)      #True or False?

In [45]:
#lets run the expressions given in the question to us:
a = 10
b = 10
print(a is b)          
print(a is not b)      
a = 1000
b = 1000
print(a is b)          
print(a is not b)         

True
False
False
True


Q10. What is the output of expression inside print statement. Cross check before running the program.
print(10+(10*32)//2**5&20+(~(-10))<<2)

In [49]:
#lets run the expression given in the question to us:
print(10+(10*32)//2**5&20+(~(-10))<<2)

20


Q11. Membership operation
in, not in are two membership operators and it returns boolean value
print('2' in 'Python2.7.8')
print(10 in [10,10.20,10+20j,'Python'])
print(10 in (10,10.20,10+20j,'Python'))
print(2 in {1,2,3})
print(3 in {1:100, 2:200, 3:300})
print(10 in range(20))Q. Membership operation

In [52]:
#lets run the expressions given in the question to us:

print('2' in 'Python2.7.8')
print(10 in [10,10.20,10+20j,'Python'])
print(10 in (10,10.20,10+20j,'Python'))
print(2 in {1,2,3})
print(3 in {1:100, 2:200, 3:300})
print(10 in range(20))

True
True
True
True
True
True


Q12. An integer can be represented in binary, octal or hexadecimal form. Declare one binary, one octal and one hexadecimal value and store them in three different variables.
Convert 9876 to its binary, octal and hexadecimal equivalent and print their corresponding value.

In [14]:
bin_num = 0b1010            #Declaring Binary value
oct_num = 0o24              #Declaring Octal value
hex_num = 0x1e              #Declaring Hexadecimal value

print('Printing declared Binary value:',bin_num)
print('Printing declared Octal value:',oct_num)
print('Printing declared Hexadecimal value:',hex_num)

print('Converting 9876 to its binary:',bin(9876))         #converting to binary
print('Converting 9876 to its Octal:',oct(9876))          #converting to octal
print('Converting 9876 to its hexadecimal:',hex(9876))    #converting to hexadecimal

Printing declared Binary value: 10
Printing declared Octal value: 20
Printing declared Hexadecimal value: 30
Converting 9876 to its binary: 0b10011010010100
Converting 9876 to its Octal: 0o23224
Converting 9876 to its hexadecimal: 0x2694


Q13. What will be the output of following:- a = 0b1010000
print(a)
b = 0o7436 print(b)
c = 0xfade print(c) print(bin(80))print(oct(3870)) print(hex(64222)) print(bin(0b1010000)) print(bin(0xfade)) print(oct(0xfade)) print(oct(0o7436)) print(hex(0b1010000)) print(hex(0xfade))

In [16]:
a = 0b1010000
print(a)

b = 0o7436 
print(b)

c = 0xfade 
print(c) 

print(bin(80))

print(oct(3870)) 

print(hex(64222)) 

print(bin(0b1010000)) 

print(bin(0xfade)) 

print(oct(0xfade)) 

print(oct(0o7436)) 

print(hex(0b1010000))

print(hex(0xfade))

80
3870
64222
0b1010000
0o7436
0xfade
0b1010000
0b1111101011011110
0o175336
0o7436
0x50
0xfade
